File: lang/ruby/src/muse/util/SequenceBuilder.rb
require "muse/util/RandomFactory"
require "muse/util/Sequences"

module SequenceBuilder

    def self.packIncreasing(arr)
        RandomFactory.seed()
        arr[0] = RandomFactory.integerN(1, 3)
        1.upto(arr.length - 1) do |i|
            arr[i] = arr[i - 1] + RandomFactory.integerN(1, 3)
        end
    end

    def self.packRandom(arr)
        RandomFactory.seed()
        arr.each_index do |i|
            arr[i] = RandomFactory.generateInteger()
        end
    end

    def self.packDecreasing(arr)
        packIncreasing(arr)
        Sequences.reverse(arr)
    end
end

File: lang/java/src/muse/util/SequenceBuilder.java
package muse.util;

public final class SequenceBuilder {

    private SequenceBuilder() {}

    public static void packIncreasing(Integer[] arr) {
        RandomFactory.seed();
        arr[0] = RandomFactory.integerN(1, 3);
        for (int i = 1, length = arr.length; i < length; i++) {
            arr[i] = arr[i - 1] + RandomFactory.integerN(1, 3);
        }
    }

    public static void packRandom(Integer[] arr) {
        RandomFactory.seed();
        for (int i = 0, length = arr.length; i < length; i++) {
            arr[i] = RandomFactory.generateInteger();
        }
    }

    public static void packDecreasing(Integer[] arr) {
        packIncreasing(arr);
        Sequences.reverse(arr);
    }
}
